sys/net/application_layer/dhcpv6/client.c

```diff
@@ -611,16 +740,19 @@ static bool _parse_reply(uint8_t *rep, size_t len)
             case DHCPV6_OPT_IA_PD:
                 ia_pd = (dhcpv6_opt_ia_pd_t *)opt;
                 break;
+            case DHCPV6_OPT_IA_NA:
+                ia_na = (dhcpv6_opt_ia_na_t *)opt;
+                break;
             case DHCPV6_OPT_SMR:
                 smr = (dhcpv6_opt_smr_t *)opt;
                 break;
             default:
                 break;
         }
     }
-    if ((cid == NULL) || (sid == NULL) || (ia_pd == NULL)) {
+    if ((cid == NULL) || (sid == NULL) || (ia_pd == NULL && ia_na == NULL)) {
```

Comment:
Rather use
```suggestion
    if ((cid == NULL) || (sid == NULL) || (ia_pd == NULL) ||
        (IS_USED(MODULE_DHCPV6_CLIENT_IA_NA) && (ia_na == NULL))) {
```
to keep both optional?

Reply:
This should now be addressed after the last rebase :)